cmd/wallet-web/src/main.js

```diff
@@ -16,11 +15,29 @@ import ToastNotificationComponent from '@/components/ToastNotification/ToastNoti
 
 const app = createApp(App);
 
-app.use(router);
-app.use(store);
-app.use(i18n);
-app.provide('polyfill', polyfill);
-app.provide('webCredentialHandler', webCredentialHandler);
-app.component('ToastNotification', ToastNotificationComponent);
+(async () => {
+  if (process.env.NODE_ENV === 'production') {
+    // call service to get the agent opts
+    const CHAPIEnabled = await axios
+      .get(`${window.location.origin}/walletconfig/agent`)
+      .then((resp) => resp.data.enableCHAPI)
```

Comment:
We already make a call to get wallet-config options before we load the first page. This adds a duplicate network call to fetch same data and adds to overall performance. Can we update code to have only one call to this API ?

Reply:
Good point, but at in `main.js` we don't yet have the Vue app initialized, so no store, router, etc. - we can only save the fetches values in local storage. However, I'm thinking we might be able to move the dynamic import/provide logic altogether to a later stage somewhere in `App.vue` and utilize the store from there. I am looking into this possibility